sdk/azcore/internal/shared/expiring_resource.go

```diff
@@ -43,13 +46,15 @@ func (er *ExpiringResource) GetResource(state interface{}) (interface{}, error)
 	// If the resource is expiring within this time window, update it eagerly.
 	// This allows other threads/goroutines to keep running by using the not-yet-expired
 	// resource value while one thread/goroutine updates the resource.
-	const window = 2 * time.Minute // This example updates the resource 2 minutes prior to expiration
+	const window = 5 * time.Minute   // This example updates the resource 5 minutes prior to expiration
+	const backoff = 30 * time.Second // Minimum wait time between eager update attempts
 
-	now, acquire, resource := time.Now(), false, er.resource
+	now, acquire, expired, resource := time.Now(), false, false, er.resource
 	// acquire exclusive lock
 	er.cond.L.Lock()
 	for {
-		if er.expiration.IsZero() || er.expiration.Before(now) {
+		expired = er.expiration.IsZero() || er.expiration.Before(now)
```

Comment:
Should the second conditional add the `window` from above to trigger a re-acquire? It won't trigger if the expiration is within 5 minutes of the expiration time (I believe).

Reply:
Yes that does answer my question, thank you!